### PR TITLE
chore(litmus-portal): Adding agent to litmus portal's subscriber and few more changes to IGC

### PR DESCRIPTION
**Pull request by rajdas98**
Changes:
- Adding go-agent for the subscriber
- Adding go-agent docker image to docker hub using Makefile
- Adding WebSocket events in igc
- Adding npm start command in package.json

- NOTE: Go-agent requires cluster-admin cluster role to apply the k8s resources

### CODE REVIEW

File: litmus-portal/backend/subscriber/agent/pkg/cluster/cluster.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	memory "k8s.io/client-go/discovery/cached"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"log"
+)
+
+var decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+
+func applyRequest(obj *unstructured.Unstructured, requestType interface{}, dr dynamic.ResourceInterface) (*unstructured.Unstructured, error) {
```
Comment (gdsoumya) [inline litmus-portal/backend/subscriber/agent/pkg/cluster/cluster.go:22]: Does this function handle any resource type[service,deployments,ns etc.] or specific resources only?

Comment (rajdas98) [reply litmus-portal/backend/subscriber/agent/pkg/cluster/cluster.go:22]: Every resource type, its a dynamic client

**Comment by rajdas98** [comment]
Subscriber Manifest
```---

apiVersion: apps/v1
kind: Deployment
metadata:
  name: subscriber
  namespace: litmus
  labels:
    app: subscriber
spec:
  replicas: 1
  selector:
    matchLabels:
      app: subscriber
  template:
    metadata:
      labels:
        app: subscriber
    spec:
      serviceAccountName: litmus-portal
      containers:
        - name: igc
          image: 'imrajdas/igc:latest'
          imagePullPolicy: Always
          env:
            - name: GQL_SERVER
              value: 'litmusportal-server-service:9002/query'
        - name: go-agent
          imagePullPolicy: Always
          image: 'imrajdas/portal-agent:latest'
```

**Comment by rajdas98** [comment]
cc @gdsoumya Take this subscriber ref for the self cluster connection

**Comment by rajdas98** [comment]
Tested locally for the agent, igc and server

**Pull request merged**